### ISSUE TITLE
Performance degrades after half a million posts

### ISSUE THREAD

**Issue by catSIXe** [rr-/szurubooru#463]
The board was very responsive up to 300-400K posts and now it takes ages to load. Is there any way to increase the Worker Count? the entire board seems to be synchronous and locking up. I can see that the waitress is very slow at processing everything.

Especially when i want to apply tags to many images.

Sometimes i get a

```
Request has been terminated
Possible causes: the network is offline, Origin is not allowed by Access-Control-Allow-Origin, the page is being unloaded, etc.
```

when i want to use the rest api for adding tags, 
`'Failed to connect to szurubooru REST API'` appears after some time.

```
22-02-08 21:40:11] waitress.queue Task queue depth is 24
server_1  | [2022-02-08 21:40:11] szurubooru.middleware.request_logger GET /post/399256 (user=shitcord, queries=8)
server_1  | [2022-02-08 21:40:11] waitress.queue Task queue depth is 24
server_1  | [2022-02-08 21:40:12] szurubooru.middleware.request_logger GET /post/522463 (user=shitcord, queries=6)
server_1  | [2022-02-08 21:40:12] waitress.queue Task queue depth is 24
```

**Comment by TiredSysOp** [comment]
Have a look [calvinthefreak](https://github.com/calvinthefreak) comments on issue #405 

> irst of all, look at your docker-compose.yml and look how your server-container is named:
> 
> version: '2'
> services:
>   server:
>     image: szurubooru/server:latest
> [..]
> Mine container is named "server" as can clearly be seen.
> 
> after this, look at your command:
> docker-compose up --scale server=5 the last statement with server=5 tells it to scale the "server" container to 5 instances.
> Optionally append the -d command for deatach.
> 
> Note on the first container like I said, only scale the server container.
> If you did not change anything to the nginx upstreamserver (I am sure, that you did not), it will be loadbalanced based on the hostname.
> 
> when you did not change anything to your docker-compose naming scheme, then just try it with the --scale server=(container count to spawn) and check if it gets any faster for multiple requests at the same time.

**Comment by catSIXe** [comment]
> Have a look [calvinthefreak](https://github.com/calvinthefreak) comments on issue #405
> 
> > irst of all, look at your docker-compose.yml and look how your server-container is named:
> > version: '2'
> > services:
> > server:
> > image: szurubooru/server:latest
> > [..]
> > Mine container is named "server" as can clearly be seen.
> > after this, look at your command:
> > docker-compose up --scale server=5 the last statement with server=5 tells it to scale the "server" container to 5 instances.
> > Optionally append the -d command for deatach.
> > Note on the first container like I said, only scale the server container.
> > If you did not change anything to the nginx upstreamserver (I am sure, that you did not), it will be loadbalanced based on the hostname.
> > when you did not change anything to your docker-compose naming scheme, then just try it with the --scale server=(container count to spawn) and check if it gets any faster for multiple requests at the same time.

This seems to somewhat improve the performance, scaling to 10 instances now. I was thinking about load balancing it before, but i weren't sure if it was supported as there is absolute no information about wether this is possible or not.

**Comment by noirscape** [comment]
I've opened a pull request to address this issue without needing people to spin up more instances by just letting you configure the amount of threads waitress is allowed to use.

Spinning up more instances *works* but is also a hack and causes the containers to start taking up more disk space on the servers. The "correct" option is to just allow waitress to use more threads (by default it uses 4, which is fine if your database accesses are still extremely quick.)

**Comment by sgsunder** [comment]
Issue should now be fixed in noirscape's merged PR, 82541536afd95ebbbb72b1983caae77b67eb8b2d.
